decidim-core/app/presenters/decidim/resource_locator_presenter.rb

```diff
@@ -62,53 +65,93 @@ def show(options = {})
     #
     # Returns a String.
     def edit(options = {})
-      admin_route_proxy.send("edit_#{member_route_name}_path", resource, options)
+      options.merge!(options_for_polymorphic)
+
+      admin_route_proxy.send("edit_#{member_route_name}_path", target, options)
     end
 
     private
 
+    def polymorphic?
+      resource.is_a? Array
+    end
+
+    def target
+      if polymorphic?
+        resource.last
+      else
+        resource
+      end
+    end
+
     # Private: Build the route to the resource.
     #
     # Returns a String.
     def member_route(route_type, options)
-      route_proxy.send("#{member_route_name}_#{route_type}", resource, options)
+      options.merge!(options_for_polymorphic)
+
+      route_proxy.send("#{member_route_name}_#{route_type}", target, options)
     end
 
     # Private: Build the route to the associated collection of resources.
     #
     # Returns a String.
     def collection_route(route_type, options)
+      options.merge!(options_for_polymorphic)
+
       route_proxy.send("#{collection_route_name}_#{route_type}", options)
     end
 
     def admin_collection_route(route_type, options)
+      options.merge!(options_for_polymorphic)
+
       admin_route_proxy.send("#{collection_route_name}_#{route_type}", options)
     end
 
     def manifest
-      resource.try(:resource_manifest) ||
-        resource.class.try(:resource_manifest) ||
-        resource.class.try(:participatory_space_manifest)
+      target.try(:resource_manifest) ||
+        target.class.try(:resource_manifest) ||
+        target.class.try(:participatory_space_manifest)
     end
 
     def component
-      resource.component if resource.respond_to?(:component)
+      target.component if target.respond_to?(:component)
```

Comment:
In the case that `resource` is an array, would you be able to retrieve the component from the first element of the `resource` array? I'm not sure, but something like this:
``` ruby
root_resource.try(:component)
```

where `root_resource` will be defined similar to target: 

```ruby
    def root_resource
      if polymorphic?
        resource.first
      else
        resource
      end
    end
```

Reply:
Just to have it clear, a nested resource does not have to `respond_to?(:component)`?

Currently I'm adding the component association to `Project` through `Budget` (not only for this case) and delegating methods to the component. 

```rb
has_one :component, through: :budget, foreign_key: "decidim_component_id", class_name: "Decidim::Component"

delegate :organization, :participatory_space, to: :component
```

I like your suggestion because this way, you can avoid adding the relation 😀

Reply:
I'd say that a resource could respond or not to `component`. But, AFAIK it's likely that the root resource has the reference to the component.
Another possible implementation would be to try it against all the resources from the array until finding a not-nil result, starting with the last one. But I think that it makes no sense to have a child resource with a reference to a component and a parent resource without it. 

Regarding the method delegation, I also think that is better not to have it, but I'm afraid that you will probably need it for other reasons.

Reply:
Thanks, I'll go for the `root_resource` and the association and delegate is no longer required for it to work 😌